nano/core_test/websocket.cpp

```diff
@@ -124,6 +124,46 @@ TEST (websocket, confirmation)
 	ASSERT_TIMELY (5s, future.wait_for (0s) == std::future_status::ready);
 }
 
+// Tests getting notification of a started election
+TEST (websocket, started_election)
+{
+	nano::system system;
+	nano::node_config config (nano::get_available_port (), system.logging);
+	config.websocket_config.enabled = true;
+	config.websocket_config.port = nano::get_available_port ();
+	auto node1 (system.add_node (config));
+
+	std::atomic<bool> ack_ready{ false };
+	auto task = ([&ack_ready, config, &node1] () {
+		fake_websocket_client client (node1->websocket_server->listening_port ());
+		client.send_message (R"json({"action": "subscribe", "topic": "started_election", "ack": "true"})json");
+		client.await_ack ();
+		ack_ready = true;
+		EXPECT_EQ (1, node1->websocket_server->subscriber_count (nano::websocket::topic::started_election));
+		return client.get_response ();
+	});
+	auto future = std::async (std::launch::async, task);
+
+	ASSERT_TIMELY (5s, ack_ready);
+
+	// Create election, causing a websocket message to be emitted
+	nano::keypair key1;
+	auto send1 (std::make_shared<nano::send_block> (nano::dev::genesis->hash (), key1.pub, 0, nano::dev::genesis_key.prv, nano::dev::genesis_key.pub, *system.work.generate (nano::dev::genesis->hash ())));
```

Comment:
This should use the block builder to make it easier to understand what the constructor pieces are.